test/cuda/CMakeLists.txt

```diff
@@ -71,17 +92,30 @@ function(add_tests PREC REQ_TOL CHECK_TOL UPSAMP)
     add_test(NAME cufinufft3d2_test_SM_${PREC}_${UPSAMP}
              COMMAND cufinufft3d_test 2 2 2 5 10 20 ${REQ_TOL} ${CHECK_TOL}
                      ${PREC} ${UPSAMP})
+
+    add_test(NAME cufinufft3d3_test_SM_${PREC}_${UPSAMP}
+             COMMAND cufinufft3d_test 2 3 2 5 10 20 ${REQ_TOL} ${CHECK_TOL}
+                     ${PREC} ${UPSAMP})
   endif()
 
   add_test(NAME cufinufft3d2_test_GM_${PREC}_${UPSAMP}
            COMMAND cufinufft3d_test 1 2 2 5 10 20 ${REQ_TOL} ${CHECK_TOL}
                    ${PREC} ${UPSAMP})
+
+  add_test(NAME cufinufft3d3_test_GM_${PREC}_${UPSAMP}
+           COMMAND cufinufft3d_test 1 3 2 5 10 20 ${REQ_TOL} ${CHECK_TOL}
+                   ${PREC} ${UPSAMP})
 endfunction()
 
+add_test(NAME cufinufft_public_api COMMAND public_api_test)
+add_test(NAME cufinufft_makeplan COMMAND test_makeplan)
+add_test(NAME cufinufft_math_test COMMAND cufinufft_math_test)
+
 add_tests(float 1e-5 2e-4 2.0)
 add_tests(double 1e-12 1e-11 2.0)
 add_tests(float 1e-5 2e-4 1.25)
 add_tests(double 1e-8 1e-7 1.25)
-
-add_test(NAME cufinufft_public_api COMMAND public_api_test)
-add_test(NAME cufinufft_makeplan COMMAND test_makeplan)
+add_tests(float 1e-5 2e-4 0.f)
```

Comment:
I'm confused how the UPSAMP=0.f string appends to the executable here? Does "f" mean float, in which case, should all the add_tests(float *) be with "f" ? If it's correct, please add comment.